src/chainstate/burn/db/burndb.rs

```diff
@@ -843,674 +648,811 @@ impl BurnDB {
+    ///    return an error.
+    pub fn get_parent_pox_id(&self, _pox_id: &PoxIdentifier) -> Result<PoxIdentifier, ()> {
+        Err(())
+    }
+    /// does the given pox identifier describe a child fork of the parent pox identifier? this returns true if they
+    ///   are equal or child is a descendant.
+    pub fn is_pox_id_a_child(&self, parent: &PoxIdentifier, child: &PoxIdentifier) -> bool {
+        return parent == child || self.get_parent_pox_id(child).as_ref() == Ok(parent)
+    }
+    pub fn stubbed() -> PoxDB {
+        PoxDB
+    }
+}
+
+// High-level functions used by ChainsCoordinator
+impl SortitionDB {
+    pub fn get_sortition_id(&self, burnchain_header_hash: &BurnchainHeaderHash, pox_id: &PoxIdentifier) -> Result<SortitionId, ()> {
+        Ok(SortitionId::new(burnchain_header_hash, pox_id))
+    }
+    pub fn is_sortition_processed(&self, burnchain_header_hash: &BurnchainHeaderHash, pox_id: &PoxIdentifier) -> Result<bool, BurnchainError> {
+        let sort_id = SortitionId::new(burnchain_header_hash, pox_id);
+
+        match SortitionDB::get_block_snapshot(&self.conn, &sort_id) {
+            Ok(opt_sn) => Ok(opt_sn.is_some()),
+            Err(e) => Err(BurnchainError::from(e))
+        }
+    }
+
+    pub fn evaluate_sortition(&mut self, burn_header: &BurnchainBlockHeader, ops: Vec<BlockstackOperationType>,
+                              burnchain: &Burnchain, pox_id: &PoxIdentifier, pox_db: &PoxDB) -> Result<(BlockSnapshot, BurnchainStateTransition), BurnchainError> {
+        let parent_pox = pox_db.get_canonical_pox_id(&burn_header.parent_block_hash)
+            .map_err(|_e| BurnchainError::MissingParentBlock)?;
+        let parent_sort_id = SortitionId::new(&burn_header.parent_block_hash, &parent_pox);
+
+        if !pox_db.is_pox_id_a_child(&parent_pox, pox_id) {
+            return Err(BurnchainError::NonCanonicalPoxId(parent_pox, pox_id.clone()));
+        }
+
+        let mut sortition_db_handle = SortitionHandleTx::begin(self, &parent_sort_id)?;
+        let parent_snapshot = sortition_db_handle.as_conn().get_block_snapshot(&burn_header.parent_block_hash)?
+            .ok_or_else(|| {
+                warn!("Unknown block {:?}", burn_header.parent_block_hash);
+                BurnchainError::MissingParentBlock
+            })?;
+
+        let new_snapshot = sortition_db_handle.process_block_txs(
+            &parent_snapshot, burn_header, burnchain, ops)?;
+
+        // commit everything!
+        sortition_db_handle.commit()?;
+        Ok(new_snapshot)
+    }
+
+    pub fn is_stacks_block_in_sortition_set(&self, sortition_id: &SortitionId, block_to_check: &BlockHeaderHash) -> Result<bool, BurnchainError> {
+        self.index_handle(sortition_id)
+            .expects_stacks_block_in_fork(block_to_check)
+            .map_err(|e| BurnchainError::from(e))
+    }
+
+    pub fn latest_stacks_blocks_processed(&self, sortition_id: &SortitionId) -> Result<u64, BurnchainError> {
```

Comment:
Is this method used anywhere?  Also, the arrival index of a block is going to be different for each node, since arrival indexes are dependent on network conditions.

Reply:
Not yet -- it's part of the interface defined by the ChainsCoordinator (#1704

Reply:
And it should be okay that this is different for each node -- this method is going to be used to assist in determining which Stacks blocks are ready for processing.